src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1092,19 +1110,101 @@ private int getMostEfficientTool(@NotNull final Block target)
         final int required = target.getHarvestLevel(target.getDefaultState());
         int bestSlot = -1;
         int bestLevel = Integer.MAX_VALUE;
-        @NotNull final InventoryCitizen inventory = worker.getInventoryCitizen();
+        @NotNull InventoryCitizen inventory = worker.getInventoryCitizen();
+
         for (int i = 0; i < inventory.getSizeInventory(); i++)
         {
             final ItemStack item = inventory.getStackInSlot(i);
             final int level = Utils.getMiningLevel(item, tool);
+
             if (level >= required && level < bestLevel)
             {
-                bestSlot = i;
-                bestLevel = level;
+                boolean isUsable = true;
+
+                if (item != null)
+                {
+                    isUsable = verifyToolLevel(item, level);
+                }
+                if (isUsable)
+                {
+                    bestSlot = i;
+                    bestLevel = level;
+                }
             }
         }
         return bestSlot;
     }
+    
+     /**
+     * Verifies if there is one tool with an acceptable level
+     * in a worker's inventory
+     *
+     * @param tool the type of tool needed
+     * @return true if tool is acceptable
+     */
+    private boolean hasToolLevel(final String tool)
+    {
+        @NotNull InventoryCitizen inventory = worker.getInventoryCitizen();
+
+        for (int i = 0; i < inventory.getSizeInventory(); i++)
+        {
+            final ItemStack item = inventory.getStackInSlot(i);
+            final int level = Utils.getMiningLevel(item, tool);
+
+            if (item != null)
+            {
+                final boolean isTool = Utils.isTool(item, tool);
+                final boolean isUsable = verifyToolLevel(item, level);
+                if (isTool && isUsable)
+                {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    
+    
+     /**
+     * Verifies if an item has an appropriated grade
+     *
+     * @param item  the type of tool needed
+     * @param level the type of tool needed
+     * @return true if tool is acceptable
+     */
+    private boolean verifyToolLevel(final ItemStack item, int level)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "verifyToolLevel" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)